src/components/ArmyList.jsx

```diff
@@ -9,17 +9,18 @@ const Body = styled.div`
   flex-grow: 3;
 `;
 
-const ArmyList = ({factionUnits, removeFunction}) => <Body>
+const ArmyList = ({factionUnits, removeFunction, updateFunction}) => <Body>
     {Object.keys(factionUnits).map(key => (
-        <UnitCard unit={factionUnits[key]} key={key} deleteFunction={removeFunction} index={key}/>
+        <UnitCard updateFunction={updateFunction} unit={factionUnits[key]} key={key} deleteFunction={removeFunction} index={key}/>
```

Comment:
Have you tried out prettier? Using prettier + auto-lint on save in VSCode is like magic - every time you save it just makes everything look good (here it would turn this line into a multi-line tag instead of a long single line tag)